pubpub/pubpub
# feature: add node label editor; exclude disabled node types from labeling

This PR adds a new section to Pub settings that allows managers/admins to enable/disable labeling of nodes by node type, and to customize the labels of enabled node groups. Labeling is disabled by default, and custom label configuration is stored in a new column `Pubs.nodeLabels` in the database.

Disabled node groups are excluded from both the reference shortcut (currently `@`) suggestion dropdown and the reference controls menu.

_Test Plan_

1. Open the settings dashboard page for a Pub.
2. Scroll to the Block Labels section of the settings page. Labeling of all node types should be disabled/greyed out by default.
3. Enable a node group by clicking on a checkbox. Change the default label value to something unique.
4. Open the pub. Create a node of the same type of the node group you enabled, and a node of a type that remained disabled.
5. Create a reference, either by using the `@` shortcut, or the editor control bar.
6. In the dropdown, you should see **only** the node of the type you enabled in step #3. The other node should not be visible.
